enumerable.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
```

Comment:
1) enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
     Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))
``` 
       expected: 5..50
            got: [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]
```     
    
  2) enumerables #my_each when a block is given when self is a hash behaves like #each calls the given block once for each element in self


```
     Failure/Error: expect(my_each_output).to eq(each_output)
     
       expected: "[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]"
            got: "a1b2c3d4e5"
     
```

  3) enumerables #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self

```
     Failure/Error: expect(range.my_each_with_index(&block)).to eq(range.each_with_index(&block))
     
       expected: 5..50
            got: [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]
```

  4) enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self

```
     Failure/Error: expect(my_each_with_index_output).to eq(each_with_index_output)
     
       expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
            got: "Num: a, idx: 0\nNum: 1, idx: 1\nNum: b, idx: 2\nNum: 2, idx: 3\nNum: c, idx: 4\nNum: 3, idx: 5\nNum: d, idx: 6\nNum: 4, idx: 7\nNum: e, idx: 8\nNum: 5, idx: 9\n"
     
```

#
#### `my_each` should be identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)).

Reply:
we need some help with the linter errors we are having @umairarshadbutt